apps/studio/components/interfaces/Storage/AnalyticsBuckets/AnalyticsBucketDetails/ConnectTablesDialog.tsx

```diff
@@ -78,52 +152,48 @@ export const ConnectTablesDialog = ({ bucketId }: { bucketId?: string }) => {
   const { data: apiKeys } = useAPIKeysQuery({ projectRef, reveal: true })
   const { serviceKey } = getKeys(apiKeys)
 
-  const { data: tables } = useTablesQuery({
+  const { sourceId, pipeline, publication } = useAnalyticsBucketAssociatedEntities({
     projectRef,
-    connectionString: project?.connectionString,
-    includeColumns: false,
+    bucketId,
   })
+  const isEditingExistingPublication = !!publication && !!pipeline
 
-  const { data: sourcesData } = useReplicationSourcesQuery({ projectRef })
-  const sourceId = sourcesData?.sources.find((s) => s.name === projectRef)?.id
-
-  const { mutateAsync: createNamespace, isLoading: isCreatingNamespace } =
-    useIcebergNamespaceCreateMutation()
-
-  const { mutateAsync: createPublication, isLoading: isCreatingPublication } =
-    useCreatePublicationMutation()
-
-  const { mutateAsync: createDestinationPipeline, isLoading: creatingDestinationPipeline } =
-    useCreateDestinationPipelineMutation({
-      onSuccess: () => {},
-    })
+  const { data: tables } = useReplicationTablesQuery({ projectRef, sourceId })
 
+  const { mutateAsync: createPublication } = useCreatePublicationMutation()
+  const { mutateAsync: updatePublication } = useUpdatePublicationMutation()
+  const { mutateAsync: createDestinationPipeline } = useCreateDestinationPipelineMutation()
   const { mutateAsync: startPipeline } = useStartPipelineMutation()
 
-  const isConnecting = isCreatingNamespace || creatingDestinationPipeline || isCreatingPublication
-
-  const onSubmit: SubmitHandler<ConnectTablesForm> = async (values) => {
-    // [Joshen] Currently creates the destination for the analytics bucket here
-    // Which also involves creating a namespace + publication
-    // Publication name is automatically generated as {bucketId}_publication
-    // Destination name is automatically generated as {bucketId}_destination
+  const progressIndicator = isEditingExistingPublication
+    ? PROGRESS_INDICATORS.UPDATE
+    : PROGRESS_INDICATORS.CREATE
+  const progressDescription = progressIndicator.find((x) => x.step === connectingStep)
 
+  const onSubmitNewPublication: SubmitHandler<ConnectTablesForm> = async (values) => {
     if (!projectRef) return console.error('Project ref is required')
-    if (!sourceId) return toast.error('Source ID is required')
     if (!bucketId) return toast.error('Bucket ID is required')
+    if (!sourceId) return toast.error('Replication has not been enabled for your project')
 
     try {
+      setIsConnecting(true)
+
+      // Step 1: Create publication
+      setConnectingStep(1)
```

Comment:
Can we remove the magic number? would rather use constant types or maybe enum (strong MAYBE). Make it easier to read as well. thoughts?